pkg/crosscluster/logical/offline_initial_scan_processor.go

```diff
@@ -392,6 +395,9 @@ func (o *offlineInitialScanProcessor) checkpoint(
 		return nil
 	}
 	o.lastFlushTime = timeutil.Now()
+	if completedInitialScan {
```

Comment:
Shouldn't this block come before the checkpoint is emitted? Putting it after the checkpoint is sent means the processor will wait for another checkpoint after recieving all of the data before shutting done.

Reply:
Added a docstring to explain the location of this branch. essentially have to update processor state _after_ Next() processes the first checkpoint with the completed span.